File: php/webhook.php
<?php

use GuzzleHttp\Client;
use Monolog\Logger;
use Monolog\Handler\StreamHandler;
use Monolog\Formatter\LineFormatter;

require 'vendor/autoload.php';

// Env setting.
$config = parse_ini_file('config.ini');
$logger = new Logger('Webhook');
$handler = new StreamHandler($config['log_file'], Logger::INFO);
$formatter = new LineFormatter(null, null, false, true);
$handler->setFormatter($formatter);
$logger->pushHandler($handler);

/**
 * Summary.
 *
 * Send message based on Webhook.
 *
 * @since 1.0.0
 *
 * @see null
 * @link https://github.com/ExinOne/webhook-samples
 *
 * @param object $config Required. Base configurations.
 * @param object $logger Required. Logging to file.
 * @param string $data Required. Message content.
 * @param string $category Required. Message category. Default PLAIN_TEXT.
 * @return null.
 */
function send_mixin($config, $logger, $data, $category = 'PLAIN_TEXT')
{
    if (empty($data)) {
        return;
    }

    $client = new Client();

    try {
        $response = $client->post($config['webhook_url'].$config['access_token'],
            [
                'headers' => [
                    'Content-Type' => 'application/json',
                ],
                'body'  => json_encode([
                    'category' => $category,
                    'data'     => $data,
                ]),
            ]);

        $body = $response->getBody();
        $status = json_decode($body)->success;
        $logger->info($body);

        if ($status == "true"){
            $logger->info("Send mixin Successfully.");
        }
        else{
            $logger->error("Send mixin failed.");
        }

    } catch (\Exception $e) {
        $logger->error($e.getMessage());
        return;
    }
}

// call send mixin function
send_mixin($config, $logger, 'Hello World')

?>

File: go/webhook.go
package main

import (
    "bytes"
    "fmt"
    "io/ioutil"
    "net/http"
    "encoding/json"
    "os"
    "log"
)

// Configuration
type Configuration struct {
	LogFile			string
    WebhookUrl    	string
    AccessToken   	string
}

// Request post body
type PostBody struct {
	Category	string `json:"category"`
	Data		string `json:"data"`
}

// Resonse value
type Response struct {
    Success bool `json:"success"`
}

// Send message to Mixin Messenger based on Webhook.
//
// Parameters: content string, message content.
//
// Return: bool, status of sending message.
//
// Examples:
//
//  sendMixin("Hello World")
//
func sendMixin(content string) bool {

	var success = false

	// Load configuration files
	configFile, _ := os.Open("conf.json")
	defer configFile.Close()
	decoder := json.NewDecoder(configFile)
	configuration := Configuration{}
	err := decoder.Decode(&configuration)

	if err != nil {
	  fmt.Println("error:", err)
	}

	webhookUrl := configuration.WebhookUrl
    accessToken := configuration.AccessToken

    // Create logger
	fileName := configuration.LogFile
	logFile, err := os.OpenFile(fileName, os.O_WRONLY|os.O_APPEND, 0644)

	if err != nil {
	   	log.Fatalf("Failed opening file: %s", err)
	}
	defer logFile.Close()

	logger := log.New(logFile, "[Info] ", log.LstdFlags)

	// Initialize post body
	postBody := PostBody {
		Category:	"PLAIN_TEXT",
		Data:		content,
	}

    jsonStr, err := json.Marshal(postBody)

    if err != nil {
		logger.SetPrefix("[Error] ")
    	logger.Println(err)
    }

    // Initialize request
    url := fmt.Sprintf("%s%s", webhookUrl, accessToken)
    req, err := http.NewRequest("POST", url, bytes.NewBuffer(jsonStr))
    req.Header.Set("Content-Type", "application/json")

    client := &http.Client{}
    resp, err := client.Do(req)
    if err != nil {
		logger.SetPrefix("[Error] ")
    	logger.Println(err)
    }
    defer resp.Body.Close()

    header := resp.Header
    body, _ := ioutil.ReadAll(resp.Body)

    response := Response{}
    err = json.Unmarshal(body, &response)

    if err != nil {
    	logger.SetPrefix("[Error] ")
    	logger.Println(err)
	}

    success = response.Success

    logger.Println(string(body))
    logger.Println(header)

    return success
}

func main() {
	var success = false
	success = sendMixin("Hello World")

	if success {
		log.Printf("Send mixin successfully.")
	} else {
		log.Fatalf("Send mixin failed.")
	}
}